physicalplan/plan.go

```diff
@@ -144,24 +160,52 @@ func newOperator(expr parser.Expr, storage storage.Queryable, mint time.Time, ma
 	}
 }
 
-func newVectorBinaryOperator(e *parser.BinaryExpr, storage storage.Queryable, mint time.Time, maxt time.Time, step, lookbackDelta time.Duration) (model.VectorOperator, error) {
-	leftOperator, err := newCancellableOperator(e.LHS, storage, mint, maxt, step, lookbackDelta)
+func unpackVectorSelector(t *parser.MatrixSelector) (*parser.VectorSelector, []*labels.Matcher, error) {
+	switch t := t.VectorSelector.(type) {
+	case *parser.VectorSelector:
+		return t, nil, nil
+	case *logicalplan.FilteredSelector:
+		return t.VectorSelector, t.Filters, nil
+	default:
+		return nil, nil, parse.ErrNotSupportedExpr
+	}
+}
+
+func newVectorSelector(selector engstore.SeriesSelector, mint time.Time, maxt time.Time, offset, lookbackDelta time.Duration, step time.Duration) (model.VectorOperator, error) {
```

Comment:
What's the difference between this and `scan.NewVectorSelector`? They are all called `VectorSelector`